ritasamir/assignment4/CountingIslands.java

```diff
@@ -0,0 +1,69 @@
+package assignment_4;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+
+import assignment_3.Point;
+
+public class CountingIslands {
+	int rowNumber;
+	int columnNumber;
+
+	/**
+	 * @param tiles
+	 *            is a 2-dimensional array of booleans, where false means water
+	 *            and true means land
+	 * @param rowNumber
+	 *            is the number of rows
+	 * @param columnNumber
+	 *            is the number of columns
+	 * @return the number of islands.
+	 */
+	public int getNumberOfIslands(boolean[][] tiles, int rowNumber,

```

Comment:
You can make this method static and then pass rowNumber and columnNumber as parameters to other methods (or better, just use the size of the boolean array instead).